java-frontend/src/test/java/org/sonar/java/cfg/VariableReadExtractorTest.java

```diff
@@ -35,13 +39,30 @@ private static MethodTree buildMethodTree(String methodCode) {
     return (MethodTree) ((ClassTree) cut.types().get(0)).members().get(2);
   }
 
-  @Test
-  void should_extract_local_vars_read() {
-    MethodTree methodTree = buildMethodTree("void foo(boolean a) { new Object() { void foo() { System.out.println(a);} };  }");
+  @ParameterizedTest(name = "[{index}] With includeFields={1}, {2} read variable should be extracted from method code: {0}]")
+  @MethodSource("provideExtractionTest")
+  void should_extract_correctly(String methodCode, boolean includeFields, int size) {
```

Comment:
This is the only refactoring that went not smoothly. 
I started with a @CsvSource, and the result was kind of messy, with string on multiple lines, it was not so clear to separate the differents values.
I then used a @MethodSource and I feel that it's much better, even with the comments that were originally present in the different tests, the result is good.